### PR TITLE
FLPATH-364: prebuilt vm onboarding tasks

### PR DESCRIPTION
**Pull request by RichardW98**
**What this PR does / why we need it**:
prebuilt vm onboarding tasks
**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story (FLPATH-xxxx)*:
Fixes #364

**Change type**
- [ ] New feature
- [ ] Bug fix
- [ ] Unit tests
- [ ] Integration tests
- [ ] CI
- [ ] Documentation
- [ ] Auto-generated SDK code

**Impacted services**
- [ ] Workflow Service
- [ ] Notification Service

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.

### CODE REVIEW

File: workflow-examples/src/main/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleVMCreationWorkFlowCheckerTask.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			vmType = getRequiredParameterValue("VM_TYPE");
+		}
+		catch (MissingParameterException e) {
+			log.error("parameter VM_TYPE is not found");
+		}
+
+		String message = null;
+		String envVars = responseDTO.getExtraVars();
+		if ("WINDOWS".equalsIgnoreCase(vmType)) {
+			String vmUsername = envVars.split("\\\"win_admin_user\\\": \\\"")[1].split("\\\"")[0];
```
Comment (gciavarrini) [inline workflow-examples/src/main/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleVMCreationWorkFlowCheckerTask.java:58]: Can you please double check this regex? Maybe the regex can be simplified as 
`envVars.split("\"win_admin_user\": \"")`

File: workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/task/AapCreateVMWorkFlowTaskTest.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+	private static final String USERNAME_TEST = "username-test";
+
+	public static final String PASSWORD_TEST = "password-test";
+
+	public static final String VM_TYPE_PARAMETER_NAME = "VM_TYPE";
+
+	public static final String VM_TYPE_PARAMETER_VALUE_TEST = "vm-type-test";
+
+	public static final String AAP_GET_JOB_RESPONSE_DTO_JOB_ID_TEST = "job-id-test";
```
Comment (gciavarrini) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/task/AapCreateVMWorkFlowTaskTest.java:45]: why public? can these vars be private?

File: workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/task/AapInstallToolsWorkFlowTaskTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+	private static final String USERNAME_TEST = "username-test";
+
+	public static final String PASSWORD_TEST = "password-test";
+
+	public static final String VM_TYPE_PARAMETER_NAME = "VM_TYPE";
+
+	public static final String VM_TYPE_PARAMETER_VALUE_TEST = "vm-type-test";
+
+	public static final String AAP_GET_JOB_RESPONSE_DTO_JOB_ID_TEST = "job-id-test";
```
Comment (gciavarrini) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/task/AapInstallToolsWorkFlowTaskTest.java:43]: why public? can these vars be private?

File: workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleVMCreationWorkFlowCheckerTaskTest.java [hunk 4]
```diff
@@ -32,20 +36,24 @@ public class AnsibleCompletionWorkFlowCheckerTaskTest extends BaseWorkFlowChecke
...
 
+	public static final String VM_TYPE_PARAMETER_NAME = "VM_TYPE";
+
 	public static final String JOB_ID_PARAMETER_VALUE_TEST = "job-id-test";
 
+	public static final String VM_TYPE_PARAMETER_VALUE_TEST = "windows";
+
 	public static final String AAP_GET_JOB_RESPONSE_DTO_STATUS_SUCCESSFUL_TEST = "successful";
 
 	public static final String AAP_GET_JOB_RESPONSE_DTO_STATUS_PENDING_TEST = "pending";
```
Comment (gciavarrini) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleVMCreationWorkFlowCheckerTaskTest.java:47]: why public? can these vars be private?

File: workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleJobCompletionWorkFlowCheckerTaskTest.java [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+	public static final String PASSWORD_TEST = "password-test";
+
+	public static final String JOB_ID_PARAMETER_NAME = "JOB_ID";
+
+	public static final String JOB_ID_PARAMETER_VALUE_TEST = "job-id-test";
+
+	public static final String AAP_GET_JOB_RESPONSE_DTO_STATUS_SUCCESSFUL_TEST = "successful";
+
+	public static final String AAP_GET_JOB_RESPONSE_DTO_STATUS_PENDING_TEST = "pending";
```
Comment (gciavarrini) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/vmonboarding/checker/AnsibleJobCompletionWorkFlowCheckerTaskTest.java:41]: why public? can these vars be private?

File: prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/ansible/AapLaunchJobWorkFlowTaskTest.java [hunk 6]
```diff
@@ -24,7 +23,7 @@
 import static org.mockito.Mockito.mockStatic;
 import static org.mockito.Mockito.spy;
 
-public class AapLaunchJobWorkFlowTaskTest extends BaseInfrastructureWorkFlowTaskTest {
```
[the remark this answers is not in the document]
Comment (RichardW98) [reply prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/ansible/AapLaunchJobWorkFlowTaskTest.java:27]: becuase this is in prebuilt-task,  can't read classess from workflow-examples

[the remark this answers is not in the document]
Comment (gciavarrini) [reply prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/ansible/AapLaunchJobWorkFlowTaskTest.java:27]: You're right! Thanks!

**Review by anludke — APPROVED** [review]
/lgtm

**Pull request merged**